### ISSUE TITLE
Settings menu is unreadable in the evening with default MacOS configuration

### ISSUE THREAD

**Issue by jneen** [mixxxdj/mixxx#13643]
### Bug Description

<img width="1203" alt="image" src="https://github.com/user-attachments/assets/20f4b39e-d601-49c2-87ad-ba2476939173">

Lots of the text is near-white on light-grey.

I have not set any custom QT themes, this is a bare install on MacOS. It looks like the inside of the menu is using hard-coded colours for some things and system-selected colours for others. I'd be happy with any colours as long as they're reasonably contrasted.

**Comment by ronso0** [comment]
Nope, all dialogs use the Qt / system theme.
(We considered styling the preferences per skin, but dismissed/delayed that because it's a ton of work to do it right)
Do all pref pages look like this?

How do the pages look like when you set the bright OS theme?

**Comment by ronso0** [comment]
and which macOS and Mixxx version do you use?

**Comment by jneen** [comment]
Looks like it's fixed on restart. M1 Max, Sonoma 14.6.1, iirc the latest. I think the issue likely happens when the system theme changes during runtime.

**Comment by jneen** [comment]
Yeah okay, confirmed - I can reproduce it by opening Mixxx and then swapping the system theme from System Preferences between light/dark.

**Comment by ronso0** [comment]
Okay, thanks for checking.

Is the theme updated in other apps' settings/dialogs?
If, yes I'm not sure if/how this can be fixed.
@fwcd what do you think?

**Comment by fwcd** [comment]
Does restarting Mixxx fix it? If so, it's probably just an unintended consequence of assuming that the theme doesn't change while the app is running (IIRC we do that for localization and a few other things too, probably out of convenience).

I agree that the interaction with automatic theme changes is unfortunate. I'm not familiar enough with our styling to say how much of this Qt is responsible for and how much we are, but my gut feeling is that it's probably something on Qt's side, so maybe worth searching for relevant issues there.

> Is the theme updated in other apps' settings/dialogs?

Yes, most apps I've encountered on macOS handle "hot" color scheme changes correctly (almost all Cocoa apps, Firefox, VSCode, GTK4 apps).

**Comment by ronso0** [comment]
@jneen In order to get an idea what's not updated on theme change, can you please post a screnshot of the correct preferences when starting Mixxx in dark mode?

**Comment by ronso0** [comment]
Seems to affect labels and groupboxes only, right?

Note: we might catch QEvent::PaletteChange in PreferencesPage and set the new palette on all children() manually, then repaint().

**Comment by xARSENICx** [comment]
I think the issue is up again, but in the Light Mode:

<img width="1310" height="1035" alt="Image" src="https://github.com/user-attachments/assets/a0e03ef0-4b4f-4f53-9412-195feae9e822" />

Jörg mentioned #14228 as the PR fixing preferences for qt > 6.8 in windows.

**Comment by xARSENICx** [comment]
@ronso0 to bring this into your attention

**Comment by ronso0** [comment]
Not sure if I'm of help here since I don't have access to a mac, all I could do is set up a branch that catches the `QEvent::PaletteChange` and tries to do [whatever] in order to re-set the palette on the entire dialog. not even sure if that can work.

Here's what Gemini proposes:

src/preferences/dialog/dlgpreferences.h: add `changeEvent`
```
protected:
    void changeEvent(QEvent* event) override;
```

and the implementation in src/preferences/dialog/dlgpreferences.cpp:
```
#include <QApplication>
#include <QStyle>
```

```
void DlgPreferences::changeEvent(QEvent* event) {
    // Intercept macOS system theme switches
    if (event->type() == QEvent::PaletteChange || event->type() == QEvent::ApplicationPaletteChange) {
        
        QPalette appPalette = QApplication::palette();
        
        // Force DlgPreferences itself to take the new palette
        this->setPalette(appPalette);
        
        // Loop through all pages managed by Mixxx's layout
        if (m_pStackedLayout) {
            for (int i = 0; i < m_pStackedLayout->count(); ++i) {
                QWidget* pPage = m_pStackedLayout->widget(i);
                if (pPage) {
                    // Update the page widget wrapper
                    pPage->setPalette(appPalette);
                    
                    // Recursively update all hidden child widgets within the page
                    // (Targeting QLabels, QComboBoxes, QTabWidgets etc.)
                    QList<QWidget*> children = pPage->findChildren<QWidget*>();
                    for (QWidget* pChild : children) {
                        child->setPalette(appPalette);
                        
                        // Force the Mac native engine to discard cached styling metrics
                        if (pChild->style()) {
                            pChild->style()->unpolish(child);
                            pChild->style()->polish(child);
                        }
                        pChild->update();
                    }
                }
            }
        }
    }

    // Pass the event down to QDialog base implementation
    QDialog::changeEvent(event);
}
```

Wanna give it a try?

**Comment by JoergAtGithub** [comment]
@jneen Could you try the version from https://github.com/mixxxdj/mixxx/pull/16661 - Does it fix this bug for you?
